### PR TITLE
Feature/scan 254 nested nft support

### PR DESCRIPTION
**Pull request by AzgatG**
Closes #ISSUE

**Description:**

nested nft support

### CODE REVIEW

File: hasura/hasura_metadata.json [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"path":"$","error":"invalid x-hasura-admin-secret/x-hasura-access-key","code":"access-denied"}
```
Comment (chernetsky) [inline hasura/hasura_metadata.json]: Looks like a bad schema.

File: lib/token/tokenDb.ts [hunk 2]
```diff
@@ -10,16 +10,40 @@ const TOKEN_FIELDS = [
   'owner_normalized',
   'data',
   'date_of_creation',
+  'parent_id',
 ];
 
+const NESTING_ADDRESS_PREFIX = '0xf8238ccfff8ed887463fd5e0';
+
+function isNestingAddress(address: string): boolean {
+  return address.indexOf(NESTING_ADDRESS_PREFIX) === 0 && address.length === 42;
```
Comment (chernetsky) [inline lib/token/tokenDb.ts]: Need to move magic 42 into constants.

File: lib/token/tokenDb.ts [hunk 3]
```diff
@@ -10,16 +10,40 @@ const TOKEN_FIELDS = [
   'owner_normalized',
   'data',
   'date_of_creation',
+  'parent_id',
 ];
 
+const NESTING_ADDRESS_PREFIX = '0xf8238ccfff8ed887463fd5e0';
```
Comment (chernetsky) [inline lib/token/tokenDb.ts]: Move this constant into constants.ts

File: lib/token/tokenDb.ts [hunk 4]
```diff
@@ -10,16 +10,40 @@ const TOKEN_FIELDS = [
...
+  let parent_id: null | number = null;
+
+  if (isNestingAddress(owner)) {
+    parent_id = getTokenIdFromNestingAddress(owner);
+  }
 
   return {
     ...token,
     ...stringifyFields(token, ['data']),
     owner_normalized: normalizeSubstrateAddress(owner),
```
Comment (chernetsky) [inline lib/token/tokenDb.ts:36]: Don't we need to normalize _parent_id_ address here? I don't know, just asking.

File: utils/utils.js [hunk 5]
```diff
@@ -153,6 +158,34 @@ function stringifyFields(obj, fieldsToStringify) {
...
+function getTokenIdFromNestingAddress(address) {
+  if (!isNestingAddress(address)) return null;
+
+  const tokenString = address.slice(
+    NESTING_ADDRESS_PREFIX.length + 8,
+    NESTING_ADDRESS_PREFIX.length + 25,
+  );
+
+  return parseInt(tokenString, 16) || null;
+}
```
Comment (chernetsky) [inline utils/utils.js:187]: It looks like we always should use these two functions together? Maybe it makes sence to have one function for parsing nesting address, for example `parseNestingAddress(fullAddress: string): { collectionId: string, tokenId: string }`?

**Review by sswebcoder — APPROVED** [review]

**Pull request merged**